### PR TITLE
Schema form

### LINKED ISSUE #224: ICMP should be renamed to Location

### LINKED ISSUE #225: Flatten Elements
Seems that the 4 levels should be directly accessible in the top menu.

It will be more clear for the end user

### PR DESCRIPTION
**Pull request by maasencioh**
### Wafer table
* Add actions at the begining of the table
* Splits elements in main menu (closes #225)
* Unifies the new requirements for creation with the previous table (closes #224)

Screen Shot 2022-02-03 at 15 44 06

### Wafer form
* ~Add possibility to create a form from a schema~ (deprecated)
* Add rich text editor component
* Add wrapper for react-hooks-form for rich text editor

Screen Shot 2022-02-03 at 15 46 07

### CODE REVIEW

File: front/src/components/RichTextFieldRHF.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import {
+  defaultErrorSerializer,
+  FieldProps,
+  RHFRegisterProps,
+  useCheckedFormRHFContext,
+} from '@/components/tailwind-ui';
+
+export type RichTextFieldRHFProps = Omit<
+  RichTextEditorProps,
+  'value' | 'onChange'
```
Comment (maasencioh) [inline front/src/components/RichTextFieldRHF.tsx:19]: The other components don't have this omition, is this correct?

**Review by maasencioh — COMMENTED** [review]
What is inside of `front/src/components/FormSchema` won't be used for the moment, but I think is a shame to delete all of it.

File: front/src/components/RichTextFieldRHF.tsx (hunk 1 above)
Comment (stropitek) [reply front/src/components/RichTextFieldRHF.tsx:19]: No but that seems correct. The components which work directly with those props (they use an uncontrolled `input`) use `register` from rhf's api, not `useController`. Those which use `useController` also omit some props from the base controlled component, for example `SelectFieldRHF`:

image

File: front/src/components/RichTextFieldRHF.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+> &
+  FieldProps &
+  RHFRegisterProps;
```
Comment (stropitek) [inline front/src/components/RichTextFieldRHF.tsx]: If you update components to the latest version, you'll have to intersect with RHFValidationProps as well

File: front/src/components/RichTextFieldRHF.tsx (hunk 1 above)
Comment (maasencioh) [reply front/src/components/RichTextFieldRHF.tsx:19]: perfect, thank you!

File: front/src/components/RichTextFieldRHF.tsx (hunk 2 above)
Comment (stropitek) [reply front/src/components/RichTextFieldRHF.tsx]: and you'll have to handle `rhfOptions.shouldUnregister`

You basically just have to forward it to `useController`

Comment (stropitek) [reply front/src/components/RichTextFieldRHF.tsx]: Also with latest components version
```suggestion
  RHFControllerProps;
```

**Comment by targos** [comment]
I trust Daniel gave you good answers. Is there something else you would like me to look at?

**Comment by maasencioh** [comment]
I just updated to latest zakodium-components, thanks you both 🙇

**Comment by stropitek** [comment]
@maasencioh can you give me some pointers on what you would like me to review? There's a lot of changes and I'm too unfamiliar with the project to navigate this efficiently

File: front/src/components/RichTextFieldRHF.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  useCheckedFormRHFContext,
+} from '@/components/tailwind-ui';
+
+export type RichTextFieldRHFProps = Omit<
+  RichTextEditorProps,
+  'value' | 'onChange'
+> &
+  FieldProps &
+  RHFValidationProps &
+  RHFControllerProps;
```
Comment (maasencioh) [inline front/src/components/RichTextFieldRHF.tsx:23]: Hi @stropitek , could you please just take a look to this file if your comments were proppertly assessed? thank you

Comment (stropitek) [reply front/src/components/RichTextFieldRHF.tsx:23]: LGTM

**Pull request merged**